### PR TITLE
improve the build system

### LINKED ISSUE #20: build and distribute precompiled libraries
Ref Tony Kelman's juliacon workshop:
https://www.youtube.com/watch?v=XKdKdfHB2KM&list=PLP8iPy9hna6SQPwZUDtAM59-wPzCPyD_S&index=12

### LINKED ISSUE #27: Use Homebrew.jl for dependencies on OSX
Something like this will get cfitsio, but for some reason it doesn't show up in the path. So I need to debug this somewhat. In the mean time there needs to be instructions for the user to use homebrew manually.

```julia
if is_apple()
    using Homebrew
    provides(Homebrew.HB, "homebrew/science/cfitsio", libcfitsio, os = :Darwin)
end
```

### LINKED ISSUE #29: get the PackageEvaluator.jl build working
http://pkg.julialang.org/detail/LibHealpix

It is somewhat embarrassing having a red dot next to the package name. I think I'll need to submit a PR to PackageEvaluator.jl to install `autoconf`, but I should actually get PackageEvaluator.jl running myself to see if that's all that's needed.

I'm not going to bother fixing the errors for julia 0.4 and 0.5 though because the package no longer supports those versions.

### PR DESCRIPTION
**Pull request by mweastwood**
We now have the following behavior

On OSX (tested on Travis CI)

* install `science/cfitsio` and `science/healpix` using Homebrew.jl
* if `pkg-config --version` fails, `pkg-config` is installed using Homebrew.jl
* deps/src/wrapper is compiled from source

On Xenial (tested locally)

* `libcfitsio3-dev`, `libchealpix-dev`, `libhealpix_cxx-dev` installed using apt
* deps/src/wrapper is compiled from source

On Trusty (tested on Travis CI)

* `libcfitsio3-dev` installed using apt
* `libchealpix` and `libhealpix_cxx` are built from source
* deps/src/wrapper is compiled from source

Other Linux (tested on tkelman/c6g6 while pre-compiling dependencies for the next case)

* Everything built from source

Linux with libtool missing (tested with local PackageEvaluator.jl run)

* pre-compiled libraries are downloaded from https://bintray.com/mweastwood/LibHealpix.jl/dependencies
(these libraries are built using the tkelman/c6g6 Docker image to try to make sure they work on as many linux distributions as possible)

Closes #20 
Closes #27 
Closes #29 
Obviates the need for JuliaCI/PackageEvaluator.jl#169

@tkelman I would really appreciate it if you could give this a quick review. Most of this PR was trial and error, and I generally have no idea what I'm doing here.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -20,36 +92,51 @@ libhealpix_src_directory = joinpath(BinDeps.depsdir(libchealpix), "src", "Healpi
...
+end
+
+provides(Binaries,
+         URI("https://dl.bintray.com/mweastwood/LibHealpix.jl/dependencies-v0.2.1-0.tar.gz"),
+         [libcfitsio, libchealpix, libhealpix_cxx, libhealpixwrapper],
+         SHA="c5ac81b6895d567081cfecc0e29567ff9c484a64c1448f7aaca7842e390cefff",
+         os=:Linux)
+
+# https://github.com/JuliaLang/BinDeps.jl/pull/163
+is_linux() && push!(BinDeps.defaults, BinDeps.Binaries)
```
Comment (tkelman) [inline deps/build.jl]: should pop this back when done
also only valid for x86_64

Comment (mweastwood) [reply deps/build.jl]: fixed. I'm running out of steam for now though, so I'm going to leave x86 support as a todo item and just restrict the use of the binaries to x64.

Comment (tkelman) [reply deps/build.jl]: linux x86 is pretty rarely used any more so probably not worth the trouble

Comment (mweastwood) [reply deps/build.jl]: Then it'll probably stay as a todo item for a rather long time. Thanks for your help!

**Pull request merged**